lib/Deserializer/functions.php

```diff
@@ -346,8 +348,20 @@ function functionCaller(Reader $reader, callable $func, string $namespace)
     }
 
     $funcArgs = [];
-    $func = is_string($func) && false !== strpos($func, '::') ? explode('::', $func) : $func;
-    $ref = is_array($func) ? new \ReflectionMethod($func[0], $func[1]) : new \ReflectionFunction($func);
+    if (is_array($func)) {
+        $ref = new \ReflectionMethod($func[0], $func[1]);
+    } elseif (is_string($func) && false !== strpos($func, '::')) {
+        // We have a string that should refer to a method that exists, like "MyClass::someMethod"
+        // ReflectionMethod knows how to handle that as-is
+        $ref = new \ReflectionMethod($func);
+    } elseif ($func instanceof \Closure || is_string($func)) {
+        // We have an actual Closure (a real function) or a string that is the name of a function
+        // ReflectionFunction can take either of those
+        $ref = new \ReflectionFunction($func);
+    } else {
+        throw new \InvalidArgumentException(__METHOD__.' unable to use func parameter with ReflectionMethod or ReflectionFunction.');
```

Comment:
I can't cover this line with a unit test.
My analysis is:
PHP always checks that `$func` is an actual callable that will work (it doesn't just check that `$func` has a format that looks like it should be callable, does actual check that something like "MyClass::someMethod" really is a function that exists in a class that exists.

So when we get to this "if" block, we know that `$func` is a working callable.

If it is in array form, then we call `ReflectionMethod`
If it is in string form with a "::" in it, then we also call `ReflectionMethod`
If it is a real Closure, or a string without "::" in it then we call `ReflectionFunction` (because it is an ordinary function)

And those are all the ways that a "callable" can be passed.

But if I do not have an "else" to catch Unicorns, then `phpstan` complains that `$ref` might not be defined.

I think that it is a computable task for `phpstan` to analyze that `$ref` must be defined. `phpstan` has to understand what are all the things whose union is a "callable". Then it can deduce that the if-elseif-... covers them all.